apps/actions-dependencies-updater/src/caches.mts

```diff
@@ -0,0 +1,95 @@
+import * as log from "./logger.mjs";
+import { GithubShaToVersionCache } from "./github.mjs";
+import { UpdateTransaction } from "./updater.mjs";
+
+const CACHE_DIR = ".cache/";
+const SHA_TO_VERSION_CACHE = "sha-to-version.json";
+
+export function initialize(forceRefresh: boolean, optionalId?: string) {
+  const id = optionalId ? optionalId : Date.now();
+  const cacheExists = fs.existsSync(path.join(CACHE_DIR, SHA_TO_VERSION_CACHE));
+
+  if (cacheExists && forceRefresh) {
+    try {
+      // rename .cache to .cache-<timestamp>
```

Comment:
Why don't we delete the cache rather than keeping the old one? It shouldn't be important to recover it since its cache data.

Reply:
Probably true for most people, I've found it useful.